src/util.hpp

```diff
@@ -339,6 +346,40 @@ namespace Util {
         double b = ldexp(frac, exp);
         return b;
     }
+
+#if defined(_WIN32) || defined(__x86_64__)
+    void CpuID(uint32_t leaf, uint32_t regs[4])
```

Comment:
passing an array as a parameter is misleading. this is the same as `uint32_t*`. i.e. the 4 is ignored.
If you want to enforce passing a C-array of 4, you can take a reference to the array, or use `std::array<uint32_t, 4>`

Reply:
Thanks, I'll change it to `uint32_t *`.